packages/runtime/src/components/builtin/Text/EditableText/editable-text.tsx

```diff
@@ -82,6 +85,7 @@ export const EditableText = forwardRef(function EditableText(
 
   const handleFocus = useCallback(() => {
     controller?.focus()
+    setHasBeenSelected(true)
```

Comment:
This prevents you from preserving selection where you have not previously focused. Ideally, this value would listen to "focus" and "blur" messages from the control and only preserve selection when you had currently selected the control. 

That is something I will include in the RichTextV2.